backend/src/plugins/ModActions/commands/BanCmd.ts

```diff
@@ -36,16 +45,82 @@ export const BanCmd = modActionsCmd({
     if (!user.id) {
       return sendErrorMessage(pluginData, msg.channel, `User not found`);
     }
+    const time = args["time"] ? args["time"] : null;
+    console.log(time);
 
+    const reason = formatReasonWithAttachments(args.reason, msg.attachments);
     const memberToBan = await resolveMember(pluginData.client, pluginData.guild, user.id);
+    // The moderator who did the action is the message author or, if used, the specified -mod
+    let mod = msg.member;
+    if (args.mod) {
+      if (!hasPermission(pluginData, "can_act_as_other", { message: msg, channelId: msg.channel.id })) {
+        sendErrorMessage(pluginData, msg.channel, "No permission for -mod");
+        return;
+      }
+
+      mod = args.mod;
+    }
 
+    // acquire a lock because of the needed user-inputs below (if banned/not on server)
+    const lock = await pluginData.locks.acquire(`ban-${user.id}`);
     let forceban = false;
+    const existingTempban = await pluginData.state.tempbans.findExistingTempbanForUserId(user.id);
+    const banned = await isBanned(pluginData, user.id);
     if (!memberToBan) {
-      const banned = await isBanned(pluginData, user.id);
-
       if (banned) {
-        sendErrorMessage(pluginData, msg.channel, `User is already banned`);
-        return;
+        // Abort if trying to ban user indefinitely if they are already banned indefinitely
+        if (!existingTempban && !time) {
+          sendErrorMessage(pluginData, msg.channel, `User is already banned indefinitely.`);
+          return;
+        }
+
+        // Ask the mod if we should update the existing ban
+        const alreadyBannedMsg = await msg.channel.createMessage("User is already banned, update ban?");
+        const reply = await waitForReaction(pluginData.client, alreadyBannedMsg, ["✅", "❌"], msg.author.id);
+
+        alreadyBannedMsg.delete();
```

Comment:
Could use a `.catch(noop)` (`noop` is from utils I think) so we don't get an error here if someone e.g. deletes the message before reacting.

Reply:
Also added to the previous ban->forceban upgrade